src/Storages/MergeTree/MergeTreeIndexMinMax.cpp

```diff
@@ -39,8 +39,25 @@ void MergeTreeIndexGranuleMinMax::serializeBinary(WriteBuffer & ostr) const
     {
         const DataTypePtr & type = index_sample_block.getByPosition(i).type;
         auto serialization = type->getDefaultSerialization();
-        serialization->serializeBinary(hyperrectangle[i].left, ostr);
-        serialization->serializeBinary(hyperrectangle[i].right, ostr);
+
+        if (!type->isNullable())
+        {
+            serialization->serializeBinary(hyperrectangle[i].left, ostr);
+            serialization->serializeBinary(hyperrectangle[i].right, ostr);
+        }
+        else
```

Comment:
Maybe we can serialize with some new format now. We just need to remain compatibility in deser.

Reply:
This will be the next step (already in progress), I don't want to make this patch complex.

Reply:
OK, let's do this here, and hope that it will not make review more complex (it is done as a separate patch).

Reply:
> I don't want to make this patch complex.

And the reason for this is that I want to deploy new version ASAP.

Reply:
@amosbird decided to make new format in a separate PR